trilinos@develop cmake fails: TPL_X11_NOT_FOUND=TRUE

`trilinos@develop ~adios2~amesos+amesos2~anasazi~aztec+belos+boost~cgns~chaco~complex~debug~dtk~epetra~epetraext+exodus+explicit_template_instantiation~float~fortran+hdf5~hypre~ifpack+ifpack2~intrepid~intrepid2~isorropia+kokkos~matio~mesquite~minitensor~ml+mpi+muelu~mumps+netcdf~nox~openmp~phalanx~piro+pnetcdf~python~rol~rythmos~sacado+shards~shylu+stk~stratimikos~suite-sparse~superlu~superlu-dist~teko~tempus+tpetra+zlib+zoltan+zoltan2~shared cxxstd=14 build_type=Release` is failing at CMake stage using:
* `spack@develop` (a68abc15c5b1b8266130f765d1bee6290e71fa7c from `Wed Jul 21 14:05:49 2021 +0200`)
* MacOS Catalina, GCC 10.3.0

Concrete spec: [trilinos.spec.yaml.txt](https://github.com/spack/spack/files/6859292/trilinos.spec.yaml.txt)
CMake Log: [trilinos-cmake.txt](https://github.com/spack/spack/files/6859296/trilinos-cmake.txt)

```
$> spack install -f ./trilinos.spec.yaml
...
==> Installing trilinos-develop-actveq3ebjw4vcwaed5y5jajtspejikm
==> No binary for trilinos-develop-actveq3ebjw4vcwaed5y5jajtspejikm found: installing from source
==> No patches needed for trilinos
==> trilinos: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    'cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/trilinos-develop-actveq3ebjw4vcwaed5y5jajtspejikm' '-DCMAKE_BUILD_TYPE:STRING=Release' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_FIND_FRAMEWORK:STRING=LAST' '-DCMAKE_FIND_APPBUNDLE:STRING=LAST' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=OFF' '-DCMAKE_INSTALL_RPATH:STRING=/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/trilinos-develop-actveq3ebjw4vcwaed5y5jajtspejikm/lib;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/trilinos-develop-actveq3ebjw4vcwaed5y5jajtspejikm/lib64;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/boost-1.76.0-q4xekipq2kowuia3nl2znshtuz6j7oyn/lib;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/bzip2-1.0.8-tazvpq2yzc66wvhtoleiqq4kudzhniri/lib;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/openmpi-4.1.1-epwoh2lg6czz2yvp67xpwbaivswvfvlr/lib;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/hwloc-2.5.0-r7nmoxncfukrmoszdwcmsvo6ycjiw2xy/lib;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/libxml2-2.9.10-q24cs2yfronapbnaredxotjksacddpgr/lib;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/libiconv-1.16-ja2b3xle67lib6myx5g66n3pei2zibqf/lib;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/xz-5.2.5-dqslvfqk6kpfsp7kuxfwqgbonwbyy7q2/lib;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/zlib-1.2.11-gkkkitec7qgtlxjvfo74wgsqpuonj4z7/lib;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/ncurses-6.2-l3nnaos6opal3udvsaopldssxjbqbdtj/lib;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/libevent-2.1.12-24le5nzo3jpedzfp24fifqxnclrnquha/lib;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/apple-clang-12.0.0/openssl-1.1.1k-fez3nc3ehb4vynunbsevjqwqtbtxhp2f/lib;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/hdf5-1.10.7-vy5q3u66jyqchtnrgqeftf7etldumske/lib;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/metis-5.1.0-3pszcv7nc5ww653sjh4xd5ped2nq4fqo/lib;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/netcdf-c-4.7.4-yh2hda4lusndd6uo5xhs2otmp2ptsjsb/lib;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/parallel-netcdf-1.12.2-i5e63gl3hujpr2vyhnakwmncf7ktyj2n/lib;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/netlib-lapack-3.9.1-jo3hzdnmrdlb72g3qrntuuzkj2jehrxg/lib;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/parmetis-4.0.3-cek662anzmsztp5ramovdirkfneucztw/lib' '-DCMAKE_PREFIX_PATH:STRING=/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/parmetis-4.0.3-cek662anzmsztp5ramovdirkfneucztw;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/netlib-lapack-3.9.1-jo3hzdnmrdlb72g3qrntuuzkj2jehrxg;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/netcdf-c-4.7.4-yh2hda4lusndd6uo5xhs2otmp2ptsjsb;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/parallel-netcdf-1.12.2-i5e63gl3hujpr2vyhnakwmncf7ktyj2n;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/metis-5.1.0-3pszcv7nc5ww653sjh4xd5ped2nq4fqo;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/hdf5-1.10.7-vy5q3u66jyqchtnrgqeftf7etldumske;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/apple-clang-12.0.0/cmake-3.20.5-ktgufgmcb2ivfplbmfxsno32mhbedjcb;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/boost-1.76.0-q4xekipq2kowuia3nl2znshtuz6j7oyn;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/openmpi-4.1.1-epwoh2lg6czz2yvp67xpwbaivswvfvlr;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/libevent-2.1.12-24le5nzo3jpedzfp24fifqxnclrnquha;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/apple-clang-12.0.0/openssl-1.1.1k-fez3nc3ehb4vynunbsevjqwqtbtxhp2f;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/hwloc-2.5.0-r7nmoxncfukrmoszdwcmsvo6ycjiw2xy;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/ncurses-6.2-l3nnaos6opal3udvsaopldssxjbqbdtj;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/libxml2-2.9.10-q24cs2yfronapbnaredxotjksacddpgr;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/zlib-1.2.11-gkkkitec7qgtlxjvfo74wgsqpuonj4z7;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/xz-5.2.5-dqslvfqk6kpfsp7kuxfwqgbonwbyy7q2;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/bzip2-1.0.8-tazvpq2yzc66wvhtoleiqq4kudzhniri;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/libiconv-1.16-ja2b3xle67lib6myx5g66n3pei2zibqf' '-DTrilinos_VERBOSE_CONFIGURE:BOOL=OFF' '-DTrilinos_ENABLE_TESTS:BOOL=OFF' '-DTrilinos_ENABLE_EXAMPLES:BOOL=OFF' '-DTrilinos_ENABLE_CXX11:BOOL=ON' '-DCMAKE_CXX_STANDARD:STRING=14' '-DBUILD_SHARED_LIBS:BOOL=OFF' '-DTrilinos_ENABLE_DEBUG:BOOL=OFF' '-DTPL_ENABLE_MPI:BOOL=ON' '-DCMAKE_C_COMPILER:STRING=/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/openmpi-4.1.1-epwoh2lg6czz2yvp67xpwbaivswvfvlr/bin/mpicc' '-DCMAKE_CXX_COMPILER:STRING=/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/openmpi-4.1.1-epwoh2lg6czz2yvp67xpwbaivswvfvlr/bin/mpic++' '-DCMAKE_Fortran_COMPILER:STRING=/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/openmpi-4.1.1-epwoh2lg6czz2yvp67xpwbaivswvfvlr/bin/mpif90' '-DMPI_BASE_DIR:STRING=/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/openmpi-4.1.1-epwoh2lg6czz2yvp67xpwbaivswvfvlr' '-DTrilinos_ENABLE_Amesos:BOOL=OFF' '-DTrilinos_ENABLE_Amesos2:BOOL=ON' '-DTrilinos_ENABLE_Anasazi:BOOL=OFF' '-DTrilinos_ENABLE_AztecOO:BOOL=OFF' '-DTrilinos_ENABLE_Belos:BOOL=ON' '-DTrilinos_ENABLE_Epetra:BOOL=OFF' '-DTrilinos_ENABLE_EpetraExt:BOOL=OFF' '-DTrilinos_ENABLE_Ifpack:BOOL=OFF' '-DTrilinos_ENABLE_Ifpack2:BOOL=ON' '-DTrilinos_ENABLE_Intrepid:BOOL=OFF' '-DTrilinos_ENABLE_Intrepid2:BOOL=OFF' '-DTrilinos_ENABLE_Isorropia:BOOL=OFF' '-DTrilinos_ENABLE_Kokkos:BOOL=ON' '-DTrilinos_ENABLE_MiniTensor:BOOL=OFF' '-DTrilinos_ENABLE_Mesquite:BOOL=OFF' '-DTrilinos_ENABLE_ML:BOOL=OFF' '-DTrilinos_ENABLE_MueLu:BOOL=ON' '-DTrilinos_ENABLE_NOX:BOOL=OFF' '-DTrilinos_ENABLE_Piro:BOOL=OFF' '-DTrilinos_ENABLE_Phalanx:BOOL=OFF' '-DTrilinos_ENABLE_PyTrilinos:BOOL=OFF' '-DTrilinos_ENABLE_ROL:BOOL=OFF' '-DTrilinos_ENABLE_Rythmos:BOOL=OFF' '-DTrilinos_ENABLE_Sacado:BOOL=OFF' '-DTrilinos_ENABLE_SCOREC:BOOL=OFF' '-DTrilinos_ENABLE_Shards:BOOL=ON' '-DTrilinos_ENABLE_ShyLU:BOOL=OFF' '-DTrilinos_ENABLE_STK:BOOL=ON' '-DTrilinos_ENABLE_Stokhos:BOOL=OFF' '-DTrilinos_ENABLE_Stratimikos:BOOL=OFF' '-DTrilinos_ENABLE_Teko:BOOL=OFF' '-DTrilinos_ENABLE_Tempus:BOOL=OFF' '-DTrilinos_ENABLE_Tpetra:BOOL=ON' '-DTrilinos_ENABLE_TrilinosCouplings:BOOL=OFF' '-DTrilinos_ENABLE_Zoltan:BOOL=ON' '-DTrilinos_ENABLE_Zoltan2:BOOL=ON' '-DEpetraExt_BUILD_BTF:BOOL=OFF' '-DEpetraExt_BUILD_EXPERIMENTAL:BOOL=OFF' '-DEpetraExt_BUILD_GRAPH_REORDERINGS:BOOL=OFF' '-DAmesos2_ENABLE_Basker:BOOL=OFF' '-DTrilinos_ENABLE_SEACAS:BOOL=ON' '-DTrilinos_ENABLE_SEACASExodus:BOOL=ON' '-DTrilinos_ENABLE_SEACASIoss:BOOL=ON' '-DTrilinos_ENABLE_SEACASEpu:BOOL=ON' '-DTrilinos_ENABLE_SEACASExodiff:BOOL=ON' '-DTrilinos_ENABLE_SEACASNemspread:BOOL=ON' '-DTrilinos_ENABLE_SEACASNemslice:BOOL=ON' '-DTrilinos_ENABLE_SEACASChaco:BOOL=OFF' '-DTrilinos_ENABLE_SEACASNemslice:BOOL=OFF' '-DTPL_ENABLE_BLAS:BOOL=ON' '-DBLAS_LIBRARY_NAMES:STRING=blas' '-DBLAS_LIBRARY_DIRS:STRING=/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/netlib-lapack-3.9.1-jo3hzdnmrdlb72g3qrntuuzkj2jehrxg/lib' '-DTPL_ENABLE_LAPACK:BOOL=ON' '-DLAPACK_LIBRARY_NAMES:STRING=lapack' '-DLAPACK_LIBRARY_DIRS:STRING=/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/netlib-lapack-3.9.1-jo3hzdnmrdlb72g3qrntuuzkj2jehrxg/lib' '-DTPL_ENABLE_Matio:BOOL=OFF' '-DTPL_ENABLE_Netcdf:BOOL=ON' '-DNetCDF_ROOT:STRING=/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/netcdf-c-4.7.4-yh2hda4lusndd6uo5xhs2otmp2ptsjsb' '-DTPL_ENABLE_HYPRE:BOOL=OFF' '-DTPL_ENABLE_Boost:BOOL=ON' '-DBoost_INCLUDE_DIRS:STRING=/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/boost-1.76.0-q4xekipq2kowuia3nl2znshtuz6j7oyn/include' '-DBoost_LIBRARY_DIRS:STRING=/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/boost-1.76.0-q4xekipq2kowuia3nl2znshtuz6j7oyn/lib' '-DTPL_ENABLE_HDF5:BOOL=ON' '-DHDF5_INCLUDE_DIRS:STRING=/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/hdf5-1.10.7-vy5q3u66jyqchtnrgqeftf7etldumske/include' '-DHDF5_LIBRARY_DIRS:STRING=/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/hdf5-1.10.7-vy5q3u66jyqchtnrgqeftf7etldumske/lib' '-DTPL_ENABLE_Cholmod:BOOL=OFF' '-DTPL_ENABLE_UMFPACK:BOOL=OFF' '-DTPL_ENABLE_METIS:BOOL=ON' '-DMETIS_LIBRARY_DIRS:STRING=/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/metis-5.1.0-3pszcv7nc5ww653sjh4xd5ped2nq4fqo/lib' '-DMETIS_LIBRARY_NAMES:STRING=metis' '-DTPL_METIS_INCLUDE_DIRS:STRING=/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/metis-5.1.0-3pszcv7nc5ww653sjh4xd5ped2nq4fqo/include' '-DTPL_ENABLE_ParMETIS:BOOL=ON' '-DParMETIS_LIBRARY_DIRS:STRING=/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/parmetis-4.0.3-cek662anzmsztp5ramovdirkfneucztw/lib;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/metis-5.1.0-3pszcv7nc5ww653sjh4xd5ped2nq4fqo/lib' '-DParMETIS_LIBRARY_NAMES:STRING=parmetis;metis' '-DTPL_ParMETIS_INCLUDE_DIRS:STRING=/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/parmetis-4.0.3-cek662anzmsztp5ramovdirkfneucztw/include;/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/metis-5.1.0-3pszcv7nc5ww653sjh4xd5ped2nq4fqo/include' '-DTPL_ENABLE_MUMPS:BOOL=OFF' '-DTPL_ENABLE_SCALAPACK:BOOL=OFF' '-DTPL_ENABLE_SuperLUDist:BOOL=OFF' '-DTPL_ENABLE_SuperLU:BOOL=OFF' '-DTPL_ENABLE_STRUMPACK:BOOL=OFF' '-DTPL_ENABLE_Pnetcdf:BOOL=ON' '-DTPL_Netcdf_Enables_Netcdf4:BOOL=ON' '-DTPL_Netcdf_PARALLEL:BOOL=ON' '-DPNetCDF_ROOT:STRING=/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/parallel-netcdf-1.12.2-i5e63gl3hujpr2vyhnakwmncf7ktyj2n' '-DTPL_ENABLE_Zlib:BOOL=ON' '-DTPL_ENABLE_Zlib:BOOL=ON' '-DZlib_ROOT:STRING=/Users/lpeyrala/exawind-collaboration/2021-07-21/spack/opt/spack//darwin-catalina-x86_64/gcc-10.3.0/zlib-1.2.11-gkkkitec7qgtlxjvfo74wgsqpuonj4z7' '-DTPL_ENABLE_CGNS:BOOL=OFF' '-DTPL_ENABLE_ADIOS2:BOOL=OFF' '-DTPL_ENABLE_HWLOC:BOOL=ON' '-DTPL_ENABLE_CUDA:BOOL=OFF' '-DTrilinos_ENABLE_OpenMP:BOOL=OFF' '-DTrilinos_ENABLE_EXPLICIT_INSTANTIATION:BOOL=ON' '-DTeuchos_ENABLE_COMPLEX:BOOL=OFF' '-DTeuchos_ENABLE_FLOAT:BOOL=OFF' '-DTpetra_INST_DOUBLE:BOOL=ON' '-DTpetra_INST_COMPLEX_DOUBLE:BOOL=OFF' '-DTpetra_INST_COMPLEX_FLOAT:BOOL=OFF' '-DTpetra_INST_FLOAT:BOOL=OFF' '-DTpetra_INST_SERIAL:BOOL=ON' '-DTpetra_INST_INT_INT:BOOL=OFF' '-DTpetra_INST_INT_LONG:BOOL=ON' '-DTpetra_INST_INT_LONG_LONG:BOOL=OFF' '-DTrilinos_ENABLE_FEI:BOOL=OFF' '-DCMAKE_MACOSX_RPATH:BOOL=ON' '-DCMAKE_CXX_FLAGS:STRING=-DSTK_NO_BOOST_STACKTRACE' '-DTrilinos_ENABLE_Pike:BOOL=OFF' '/var/folders/pw/kz96tjsx4fd57wpfj306swhh0000gp/T/lpeyrala/spack-stage/spack-stage-trilinos-develop-actveq3ebjw4vcwaed5y5jajtspejikm/spack-src'

1 error found in build log:
     681    downstream packages that have required dependencies on it, including
     682    the package 'SEACASSVDI' which triggered its enable.
     683    When you reconfigure, just grep the cmake stdout for 'X11'
     684    and then follow the disables that occur as a result to see what impact
     685    this TPL disable has on the configuration of Trilinos.
     686
  >> 687    CMake Error at cmake/tribits/core/package_arch/TribitsProcessEnabledTpl.cmake:144 (MESSAGE):
     688      ERROR: TPL_X11_NOT_FOUND=TRUE, aborting!
     689    Call Stack (most recent call first):
     690      cmake/tribits/core/package_arch/TribitsGlobalMacros.cmake:1611 (TRIBITS_PROCESS_ENABLED_TPL)
     691      cmake/tribits/core/package_arch/TribitsProjectImpl.cmake:207 (TRIBITS_PROCESS_ENABLED_TPLS)
     692      cmake/tribits/core/package_arch/TribitsProject.cmake:93 (TRIBITS_PROJECT_IMPL)
     693      CMakeLists.txt:109 (TRIBITS_PROJECT)

See build log for details:
  /var/folders/pw/kz96tjsx4fd57wpfj306swhh0000gp/T/lpeyrala/spack-stage/spack-stage-trilinos-develop-actveq3ebjw4vcwaed5y5jajtspejikm/spack-build-out.txt
```

@sethrj in case you know this one too :)

--- a/var/spack/repos/builtin/packages/trilinos/package.py
+++ b/var/spack/repos/builtin/packages/trilinos/package.py
@@ -252,6 +252,8 @@ class Trilinos(CMakePackage, CudaPackage):
              placement='packages/mesquite',
              when='+mesquite @develop')
 
+    # ###################### Conflicts ##########################
+
     # Epetra packages
     with when('~epetra'):
         conflicts('+amesos')
@@ -297,12 +299,12 @@ class Trilinos(CMakePackage, CudaPackage):
     conflicts('+tempus', when='~nox')
     conflicts('+zoltan2', when='~zoltan')
 
-    # Only allow DTK with Trilinos 12.14 and develop
+    # Only allow DTK with Trilinos 12
     conflicts('+dtk', when='~boost')
     conflicts('+dtk', when='~intrepid2')
-    conflicts('+dtk', when='@0:12.12.99,master')
+    conflicts('+dtk', when='@0:12.12.99,develop,master')
 
-    # Only allow Mesquite with Trilinos 12.12 and up, and develop
+    # Only allow Mesquite with older Trilinos 12.12 up to 13
     conflicts('+mesquite', when='@0:12.10.99,master,develop')
     # Can only use one type of SuperLU
     conflicts('+superlu-dist', when='+superlu')
@@ -325,10 +327,10 @@ class Trilinos(CMakePackage, CudaPackage):
         '+shared', when='+stk platform=darwin',
         msg='Cannot build Trilinos with STK as a shared library on Darwin.'
     )
-    # ADIOS2 was only added after v12.14.1
     conflicts('+adios2', when='@:12.14.1')
     conflicts('+adios2', when='@xsdk-0.2.0')
     conflicts('+pnetcdf', when='~netcdf')
+    conflicts('+pnetcdf', when='~mpi')
     conflicts('+cuda_rdc', when='~cuda')
     conflicts('+wrapper', when='~cuda')
     conflicts('+wrapper', when='%clang')
@@ -358,26 +360,23 @@ class Trilinos(CMakePackage, CudaPackage):
 
     # ###################### Dependencies ##########################
 
+    # Explicit dependency variants
+    depends_on('adios2', when='+adios2')
     depends_on('blas')
-    depends_on('lapack')
     depends_on('boost', when='+boost')
+    depends_on('cgns', when='+cgns')
     depends_on('hdf5+hl', when='+hdf5')
+    depends_on('hdf5~mpi', when='+hdf5~mpi')
+    depends_on('hdf5+mpi', when="+hdf5+mpi")
+    depends_on('lapack')
     depends_on('matio', when='+matio')
+    depends_on('mpi', when='+mpi')
+    depends_on('netcdf-c+mpi+parallel-netcdf', when="+netcdf+pnetcdf@master,12.12.1:")
+    depends_on('netcdf-c+mpi', when="+netcdf~pnetcdf+mpi")
+    depends_on('netcdf-c', when="+netcdf")
     depends_on('suite-sparse', when='+suite-sparse')
     depends_on('zlib', when="+zlib")
 
-    # MPI related dependencies
-    depends_on('mpi', when='+mpi')
-    depends_on('hdf5+mpi', when="+hdf5+mpi")
-    depends_on('netcdf-c', when="+netcdf")
-    depends_on('netcdf-c+mpi', when="+netcdf~pnetcdf+mpi")
-    depends_on('netcdf-c+mpi+parallel-netcdf', when="+netcdf+pnetcdf@master,12.12.1:")
-    depends_on('parallel-netcdf', when="+netcdf+pnetcdf@master,12.12.1:")
-    depends_on('metis', when='+zoltan')
-    depends_on('parmetis', when='+mpi +zoltan')
-    depends_on('parmetis', when='+scorec')
-    depends_on('cgns', when='+cgns')
-    depends_on('adios2', when='+adios2')
     # Trilinos' Tribits config system is limited which makes it very tricky to
     # link Amesos with static MUMPS, see
     # https://trilinos.org/docs/dev/packages/amesos2/doc/html/classAmesos2_1_1MUMPS.html
@@ -410,9 +409,12 @@ class Trilinos(CMakePackage, CudaPackage):
     depends_on('hwloc', when='@13: +kokkos')
     depends_on('hwloc+cuda', when='@13: +kokkos+cuda')
 
-    # Dependencies/conflicts when MPI is disabled
-    depends_on('hdf5~mpi', when='+hdf5~mpi')
-    conflicts('+pnetcdf', when='~mpi')
+    # Variant requirements from packages
+    depends_on('metis', when='+zoltan')
+    depends_on('parmetis', when='+mpi +zoltan')
+    depends_on('parmetis', when='+scorec')
+
+    # ###################### Patches ##########################
 
     patch('umfpack_from_suitesparse.patch', when='@11.14.1:12.8.1')
     patch('xlf_seacas.patch', when='@12.10.1:12.12.1 %xl')
@@ -425,19 +427,19 @@ class Trilinos(CMakePackage, CudaPackage):
     patch('cray_secas_12_12_1.patch', when='@12.12.1%cce')
     patch('cray_secas.patch', when='@12.14.1:%cce')
 
+    # workaround an NVCC bug with c++14 (https://github.com/trilinos/Trilinos/issues/6954)
+    # avoid calling deprecated functions with CUDA-11
+    patch('fix_cxx14_cuda11.patch', when='@13.0.0:13.0.1 cxxstd=14 ^cuda@11:')
+    # Allow building with +teko gotype=long
+    patch('https://github.com/trilinos/Trilinos/commit/b17f20a0b91e0b9fc5b1b0af3c8a34e2a4874f3f.patch',
+          sha256='dee6c55fe38eb7f6367e1896d6bc7483f6f9ab8fa252503050cc0c68c6340610',
+          when='@13.0.0:13.0.1 +teko gotype=long')
+
     def flag_handler(self, name, flags):
         if self.spec.satisfies('%cce'):
             if name == 'ldflags':
                 flags.append('-fuse-ld=gold')
         return (None, None, flags)
-
-    # workaround an NVCC bug with c++14 (https://github.com/trilinos/Trilinos/issues/6954)
-    # avoid calling deprecated functions with CUDA-11
-    patch('fix_cxx14_cuda11.patch', when='@13.0.0:13.0.1 cxxstd=14 ^cuda@11:')
-
-    patch('https://github.com/trilinos/Trilinos/commit/b17f20a0b91e0b9fc5b1b0af3c8a34e2a4874f3f.patch',
-          sha256='dee6c55fe38eb7f6367e1896d6bc7483f6f9ab8fa252503050cc0c68c6340610',
-          when='@13.0.0:13.0.1 +teko gotype=long')
 
     def url_for_version(self, version):
         url = "https://github.com/trilinos/Trilinos/archive/trilinos-release-{0}.tar.gz"

--- a/var/spack/repos/builtin/packages/trilinos/package.py
+++ b/var/spack/repos/builtin/packages/trilinos/package.py
@@ -411,6 +411,7 @@ class Trilinos(CMakePackage, CudaPackage):
 
     # Variant requirements from packages
     depends_on('metis', when='+zoltan')
+    depends_on('libx11', when='+exodus')
     depends_on('parmetis', when='+mpi +zoltan')
     depends_on('parmetis', when='+scorec')
 
@@ -610,41 +611,35 @@ class Trilinos(CMakePackage, CudaPackage):
 
         # ######################### TPLs #############################
 
-        blas = spec['blas'].libs
-        lapack = spec['lapack'].libs
-        options.extend([
-            define('TPL_ENABLE_BLAS', True),
-            define('BLAS_LIBRARY_NAMES', blas.names),
-            define('BLAS_LIBRARY_DIRS', blas.directories),
-            define('TPL_ENABLE_LAPACK', True),
-            define('LAPACK_LIBRARY_NAMES', lapack.names),
-            define('LAPACK_LIBRARY_DIRS', lapack.directories),
-            define_tpl_enable('Matio'),
-        ])
-
-        options.append(define_tpl_enable('Netcdf'))
-        if '+netcdf' in spec:
-            options.append(define('NetCDF_ROOT', spec['netcdf-c'].prefix))
-
-        options.append(define_tpl_enable('HYPRE'))
-        if '+hypre' in spec:
+        # Enable TPLs based on whether they're in our spec, not whether they're
+        # variant names: packages/features should disable availability
+        for tpl_name, dep_name in [
+            ('ADIOS2', 'adios2'),
+            ('BLAS', 'blas'),
+            ('Boost', 'boost'),
+            ('CGNS', 'cgns'),
+            ('HDF5', 'hdf5'),
+            ('HYPRE', 'hypre'),
+            ('HWLOC', 'hwloc'),
+            ('LAPACK', 'lapack'),
+            ('Matio', 'matio'),
+            ('METIS', 'metis'),
+            ('Netcdf', 'netcdf-c'),
+            ('STRUMPACK', 'strumpack'),
+            ('SuperLU', 'superlu'),
+            ('X11', 'libx11'),
+            ('Zlib', 'zlib'),
+        ]:
+            have_dep = (dep_name in spec)
+            options.append(define('TPL_ENABLE_' + tpl_name, have_dep))
+            if not have_dep:
+                continue
+            depspec = spec[dep_name]
             options.extend([
-                define('HYPRE_INCLUDE_DIRS', spec['hypre'].prefix.include),
-                define('HYPRE_LIBRARY_DIRS', spec['hypre'].prefix.lib),
-            ])
-
-        options.append(define_tpl_enable('Boost'))
-        if '+boost' in spec:
-            options.extend([
-                define('Boost_INCLUDE_DIRS', spec['boost'].prefix.include),
-                define('Boost_LIBRARY_DIRS', spec['boost'].prefix.lib),
-            ])
-
-        options.append(define_tpl_enable('HDF5'))
-        if '+hdf5' in spec:
-            options.extend([
-                define('HDF5_INCLUDE_DIRS', spec['hdf5'].prefix.include),
-                define('HDF5_LIBRARY_DIRS', spec['hdf5'].prefix.lib),
+                define('TPL_' + tpl_name + '_INCLUDE_DIRS', depspec.prefix.include),
+                define(tpl_name + '_ROOT', depspec.prefix),
+                define(tpl_name + '_LIBRARY_NAMES', depspec.libs.names),
+                define(tpl_name + '_LIBRARY_DIRS', depspec.libs.directories),
             ])
 
         if '+suite-sparse' in spec:
@@ -732,22 +727,8 @@ class Trilinos(CMakePackage, CudaPackage):
                     define('HAVE_SUPERLUDIST_LUSTRUCTINIT_2ARG', True),
                 ])
 
-        options.append(define_tpl_enable('SuperLU'))
-        if '+superlu' in spec:
-            options.extend([
-                define('SuperLU_LIBRARY_DIRS', spec['superlu'].prefix.lib),
-                define('SuperLU_INCLUDE_DIRS', spec['superlu'].prefix.include),
-            ])
-
-        options.append(define_tpl_enable('STRUMPACK'))
         if '+strumpack' in spec:
-            options.extend([
-                define('Amesos2_ENABLE_STRUMPACK', True),
-                define('STRUMPACK_LIBRARY_DIRS',
-                       spec['strumpack'].libs.directories[0]),
-                define('STRUMPACK_INCLUDE_DIRS',
-                       spec['strumpack'].headers.directories[0]),
-            ])
+            options.append(define('Amesos2_ENABLE_STRUMPACK', True))
 
         options.append(define_tpl_enable('Pnetcdf'))
         if '+pnetcdf' in spec:
@@ -757,25 +738,7 @@ class Trilinos(CMakePackage, CudaPackage):
                 define('PNetCDF_ROOT', spec['parallel-netcdf'].prefix),
             ])
 
-        options.append(define_tpl_enable('Zlib'))
-        if '+zlib' in spec:
-            options.extend([
-                define('TPL_ENABLE_Zlib', True),
-                define('Zlib_ROOT', spec['zlib'].prefix),
-            ])
-
-        options.append(define_tpl_enable('CGNS'))
-        if '+cgns' in spec:
-            options.extend([
-                define('TPL_ENABLE_CGNS', True),
-                define('CGNS_INCLUDE_DIRS', spec['cgns'].prefix.include),
-                define('CGNS_LIBRARY_DIRS', spec['cgns'].prefix.lib),
-            ])
-
-        options.append(define_from_variant('TPL_ENABLE_ADIOS2', 'adios2'))
-
         if '@13: +kokkos' in spec:
-            options.append(define('TPL_ENABLE_HWLOC', True))
             kkmarch = Kokkos.spack_micro_arch_map.get(spec.target.name, None)
             if kkmarch:
                 options.append(define("Kokkos_ARCH_" + kkmarch.upper(), True))

